packages/compiler/test/checker/augment-decorators.test.ts

```diff
@@ -170,6 +171,74 @@ describe("compiler: checker: augment decorators", () => {
     it("interface", () => expectTarget(`@test("target") interface Foo { }`, "Foo"));
     it("operation in interface", () =>
       expectTarget(`interface Foo { @test("target") list(): void }`, "Foo.list"));
+    it("uninstantiated template", async () => {
+      testHost.addJsFile("test.js", {
+        $customName(_: any, t: Type, n: string) {
+          const runOnTarget: Type | undefined = t;
+          const customName: string | undefined = n;
+          if (runOnTarget) {
+          }
+          if (customName) {
+          }
+        },
+      });
+
+      testHost.addTypeSpecFile(
+        "test.tsp",
+        `
+          import "./test.js";
+  
+          model Foo<T> {
+            testProp: T;
+          };
+  
+          @test
+          op stringTest(): Foo<string>;
+  
+          @@customName(Foo, "Some foo thing");
+          `
+      );
```

Comment:
@timotheeguerin so this will _not_ work if you do:
`@@customName(Foo<T>, "Some foo thing");`

Is that okay? One scenario we've seen with docs looks like:
```
@doc("Foo of {T}", T)
model Foo<T> {
  prop: T;
}
```

How would that work with augment? Like this?
`@@doc(Foo<T>, "Foo of {T}", T)`

With this PR, the above would throw a diagnostic.